dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/get-fee-record-correction-review.controller/helpers.test.ts

```diff
@@ -0,0 +1,237 @@
+      // Arrange
+      const reason = RECORD_CORRECTION_REASON.FACILITY_ID_INCORRECT;
+      const facilityId = 'some-value';
+      const formData = {
+        facilityId,
+      };
+
+      // Act
+      const formattedValue = getFormattedFormDataValueForCorrectionReason(formData, reason);
+
+      // Assert
+      expect(formattedValue).toEqual(facilityId);
+    });
+
+    it(`should map form data value for reason "${RECORD_CORRECTION_REASON.REPORTED_CURRENCY_INCORRECT}" to original value`, () => {
+      // Arrange
+      const reason = RECORD_CORRECTION_REASON.REPORTED_CURRENCY_INCORRECT;
+      const reportedCurrency = CURRENCY.GBP;
+      const formData = {
+        reportedCurrency,
+      };
+
+      // Act
+      const formattedValue = getFormattedFormDataValueForCorrectionReason(formData, reason);
+
+      // Assert
+      expect(formattedValue).toEqual(reportedCurrency);
+    });
+
+    it(`should map form data value for reason "${RECORD_CORRECTION_REASON.REPORTED_FEE_INCORRECT}" to formatted monetary value`, () => {
+      // Arrange
+      const reason = RECORD_CORRECTION_REASON.REPORTED_FEE_INCORRECT;
+      const reportedFee = 123.45;
+      const formData = {
+        reportedFee,
+      };
+
+      // Act
+      const formattedValue = getFormattedFormDataValueForCorrectionReason(formData, reason);
+
+      // Assert
+      expect(formattedValue).toEqual(getFormattedMonetaryValue(reportedFee));
+    });
+
+    it(`should map form data value for reason "${RECORD_CORRECTION_REASON.UTILISATION_INCORRECT}" to formatted monetary value`, () => {
+      // Arrange
+      const reason = RECORD_CORRECTION_REASON.UTILISATION_INCORRECT;
+      const utilisation = 10000.23;
+      const formData = {
+        utilisation,
+      };
+
+      // Act
+      const formattedValue = getFormattedFormDataValueForCorrectionReason(formData, reason);
+
+      // Assert
+      expect(formattedValue).toEqual(getFormattedMonetaryValue(utilisation));
+    });
+
+    it(`should map form data value for reason "${RECORD_CORRECTION_REASON.OTHER}" to a hyphen character`, () => {
+      // Arrange
+      const reason = RECORD_CORRECTION_REASON.OTHER;
+      const formData = {
+        additionalComments: 'Some additional bank comments',
+      };
+
+      // Act
+      const formattedValue = getFormattedFormDataValueForCorrectionReason(formData, reason);
+
+      // Assert
+      expect(formattedValue).toEqual('-');
+    });
+  });
+
+  describe('mapFormDataToFormattedValues', () => {
+    it('should return an empty array if no reasons are provided', () => {
+      // Arrange
+      const reasons: RecordCorrectionReason[] = [];
+      const formData = {};
+
+      // Act
+      const formattedValues = mapFormDataToFormattedValues(formData, reasons);
+
+      // Assert
+      expect(formattedValues).toEqual([]);
+    });
+
+    it(`should return the expected array of formatted form data values when all reasons are provided`, () => {
```

Comment:
Could we also have a test for 'some reasons' please as this will actually be the most common usage of this function